GPU/Vulkan/FragmentShaderGeneratorVulkan.cpp

```diff
@@ -86,7 +86,9 @@ bool GenerateVulkanGLSLFragmentShader(const FShaderID &id, char *buffer, uint32_
 	bool earlyFragmentTests = ((!enableAlphaTest && !enableColorTest) || testForceToZero) && !gstate_c.Supports(GPU_ROUND_FRAGMENT_DEPTH_TO_16BIT);
 	bool hasStencilOutput = stencilToAlpha != REPLACE_ALPHA_NO || id.Bit(FS_BIT_REPLACE_ALPHA_WITH_STENCIL_TYPE) == 0;
 
-	bool isAdreno = vulkanVendorId == VULKAN_VENDOR_QUALCOMM;
+	// TODO: This is a bug affecting shader cache generality - we CANNOT check anything but the shader ID and (indirectly) the game ID in here really.
```

Comment:
Well, at least as long as `loadRawPipelineCache` is false, it doesn't harm the shader cache.  It does mean if I take the vkshadercache file and load it on a Mali device, a different shader will be generated, but the use of `gstate_c.Supports` also means this.

So this comment indicates we need to move all `gstate_c.Supports` checks to the shader id too - but why?  Just to make sure if they move the PSP/SYSTEM/CACHE folder to a different device it will... preload shaders it will never use?

I'd argue that we should just include the vendor maybe and Supports flags in vkshadercache header, if and when we enable `loadRawPipelineCache`.

Or... am I missing something?

-[Unknown]

Reply:
The case that breaks is if you toggle VendorChecksEnabled - it could accidentally use shaders from the previous state. But yeah, it can also be worked around by manually wiping the shader cache.

Reply:
Hmm, wouldn't that only be true if the id DID use the vendor?

What would happen now is that it would remember an id - say:

FS_BIT_REPLACE_ALPHA_WITH_STENCIL_TYPE = 0
stencilToAlpha = REPLACE_ALPHA_YES
enableAlphaTest = YES
testForceToZero = NO

Initially, when vendor checks are enabled, it would read this 64-bit ID and generate a shader with the vendor check.  INI setting respected.

Once toggling the setting (without wiping the shader cache), it will read this 64-bit ID and generate a shader without the vendor check.  INI setting respected.

Of course, if `loadRawPipelineCache` was `true` (it never is, currently), then presumably the `VkPipelineCacheCreateInfo::initialData` field has shader data and would indeed create the wrong shader.  But today, this is not a problem.

-[Unknown]

Reply:
Oh duh, you're right, only an issue if we actually store the binary pipelinecache, which we don't.